### ISSUE TITLE
Failed to load parser in code-blocks extension

### ISSUE THREAD

**Issue by schlace** [selfint/code-blocks#173]
Issue Description:

I am experiencing issues with the code-blocks extension. After setting up the environment and installing the parsers as prompted by the extension (for C, Python, JSON, etc.), the extension works fine initially. However, after a while or when switching files, I encounter the following error:

Failed to load parser for c: Failed to load parser for language c > Failed to load c:\Users\22016933\.vscode\extensions\selfint.code-blocks-0.6.1\parsers\tree-sitter-c\tree-sitter-c.wasm > {}

The parser fails to load. If I delete the directory and reinstall the parser, the issue persists. It works immediately after installation but fails again after some time or when switching files.

Previously, I tried manually creating the wasm file using tree-sitter build --wasm in the directory c:\Users\22016933\.vscode\extensions\selfint.code-blocks-0.6.1\parsers\tree-sitter-c\. This worked for a while, but after reinstalling the VSCode environment, this method no longer resolves the issue. I am consistently encountering the “Failed to load” error, which prevents me from using this otherwise excellent extension.

Could you please provide a solution or guidance on how to resolve this issue? Your assistance would be greatly appreciated!

Environment Information
VSCode Version: 1.92.0
Extension Version: 0.6.1
Operating System: Windows 11

Thank you for your help!

**Comment by schlace** [comment]
@selfint

**Comment by selfint** [comment]
Hey @schlace, this is a known issue, It's caused by the tree-sitter library failing to properly load/free the parser (I think).

Currently reloading VSCode ("Reload window", or directly calling `workbench.action.reloadWindow`) resolves the issue temporarily.

Hopefully I can get `node-tree-sitter` working and remove WASM dependency, which should solve this issue.

Let me know if this doesn't work for you.

**Comment by Stan-Stani** [comment]
I'm on Windows 11.
Extension Version: v0.6.1
VSCode Version: 1.95.3


I get this error:


```Failed to load parser for scss: Failed to download/build parser for language scss > Failed to build .wasm parser > shell command 'tree-sitter build-wasm ' failed > error: {"code":2,"killed":false,"signal":null,"cmd":"tree-sitter build-wasm "}, logs: "error: unrecognized subcommand 'build-wasm'\n\n  tip: some similar subcommands exist: 'i', 'b', 'build'\n\nUsage: tree-sitter.exe <COMMAND>\n\nFor more information, try '--help'.\n"```

When I first added the extension I got the TS parsing working but now anytime I try to add a parser that sort of error appears.

**Comment by selfint** [comment]
Yeah I was slow updating the command, should be resolved in the next few days.

Tracking issue: https://github.com/selfint/code-blocks/issues/183

**Comment by selfint** [comment]
Just released a new version (`v0.6.2`) that patches this issue.

It works in CI and on my machine™, let me know if you have an further issues with this.
